src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -1,130 +1,76 @@
-                }
-                return response.text();
-            } )
-            .then( data => {
-                let chapterList = [];
-                let dom = this.createDOM( data );
-                dom.querySelectorAll( 'div#list div.stream' ).forEach ( card => {
-                    let info = card.querySelector( 'div.cate a span' ).innerText.trim();
-                    let chapters = [...card.querySelectorAll( 'div.volume ul.chapter li' )].map( element => {
-                        let number = element.querySelector( 'div.tit' );
-                        number = number ? number.innerText.trim() : '';
-                        let title = element.querySelector( 'div.txt' );
-                        title = title ? title.innerText.trim() : '';
-                        let seperator = number && title ? ' ' : '';
-                        element = element.querySelector( 'div.tit a.ch' );
-                        // remove last path element => show all images on single page
-                        element.href = element.href.replace( /\/[^/]*$/, '' );
-                        return {
-                            id: this.getRelativeLink( element ),
-                            title: number + seperator + title + ` [${info}]`,
-                            language: 'en'
-                        };
-                    } );
-                    chapterList = chapterList.concat( chapters );
-                } );
-                callback( null, chapterList );
-            } )
-            .catch( error => {
-                console.error( error, manga );
-                callback( error, undefined );
-            } );
-    }
-
-    /**
-     *
-     */
-    _getPageList( manga, chapter, callback ) {
-        fetch( this.url + chapter.id, this.requestOptions )
-            .then( response => {
-                if( response.status !== 200 ) {
-                    throw new Error( `Failed to receive page list (status: ${response.status}) - ${response.statusText}` );
-                }
-                return response.text();
-            } )
-            .then( data => {
-                let pages = data.match( /_load_pages\s*=\s*(\[.*?\])\s*;/ ) || data.match( /images\s*=\s*(\[.*?\])\s*;/ );
-                let pageList = JSON.parse( pages[1] ).map( page => page['u'] ? page.u : page );
-                callback( null, pageList );
-            } )
-            .catch( error => {
-                console.error( error, chapter );
-                callback( error, undefined );
-            } );
+    async _getPages(chapter) {
+        const script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    try {
+                        resolve(app.items.map(item => item.isrc));
+                    } catch(error) {
+                        reject(error);
+                    }
+                }, 2500);
+            });
+        `;
+        const uri = new URL(chapter.id, this.url);
```

Comment:
I prefer leaving it separate for readability (nested composition).
Why nesting seems no problem for such short URLs, it might be unclear for the composition of longer URLs.
To be consistent i keep all URLs outside the `Request` constructor.
```javascript
// separate composition
const uri = new URL('/manga/list?page=' + page, this.url);
const request = new Request(uri, this.RequestOptions);
const data = await this.fetchDOM(request, 'div.mangas > a');

// nested composition
const data = await this.fetchDOM(new Request(new URL('/manga/list?page=' + page, this.url), this.RequestOptions), 'div.mangas > a');
```